### PR TITLE
Updated documentation for the v5 release

### PR DESCRIPTION
**Pull request by heath-freenome**
### Reasons for making this change

- Updated `/latest/` with `/stable/` for all `.md` files, mirroring changes made on master
- Changed all uses of `<Form schema={schema} />` in existing documentation to add the required `validator` from `@rjsf/validator-ajv6`
- Updated the `validation.md` docs to cover v5 changes and additions
- Added a new `utility-function.md` to document the `@rjsf/utils` function, constants and types
- Deleted the `customizing-material-ui.md` file since we split apart the two theme versions
- Added the version 5 migration document
- Updated `@rjsf/utils` to add or improve documentation
  - Also refactored out the `TemplateTypes` interfaces to help support future work related to expanding templates
  - Added a utility type, `MakeUIType` and refactored out `UIOptionsBaseType` in order to dry out the `UiOptionsType` and `UiSchema` types
  - Added a few more `xxx_KEY` values to the `constants.ts` file, using them as appropriate
- Updated `Form` in `@rjsf/core` to move the `IChangeEvent` type from `utils` extending it from `FormState`
  - The prop `validate` was renamed to `customValidate`
- Updated `withTheme` to add the missing generics onto `Form`

### Checklist

* [x] **I'm updating documentation**
  - [x] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [ ] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [x] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [ ] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/master/CHANGELOG.md) with a description of the PR
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: docs/5.x upgrade guide.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+Unfortunately, there is required work pending to properly support React 18, so use with it at your own risk.
+
+### New packages
+
+There are three new packages added in RJSF version 5:
+- `@rjsf/utils`: All of the [utility functions](https://react-jsonschema-form.readthedocs.io/en/stable/api-reference/utiltity-functions) previously imported from `@rjsf/core/utils` as well as the Typescript types for RJSV version 5.
+- `@rjsf/validator-ajv6`: The [ajv](https://github.com/ajv-validator/ajv)-v6-based validator refactored out of `@rjsf/core@4.x`, that implements the `ValidatorType` interface defined in `@rjsf/utils`.
+- `@rjsf/mui`: Previously `@rjsf/material-ui/v5`, now provided as its own theme.
+
+### `@rjsf/core` breaking changes
```
Comment (heath-freenome) [inline docs/5.x upgrade guide.md:29]: All Caps

File: docs/5.x upgrade guide.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+#### validate renamed
+In version 5, the `validate` prop on `Form` was renamed to `customValidate` to avoid confusion with the new `validator` prop. 
+
+#### Utils
+In version 5, all the utility functions that were previously accessed via `import { utils } from '@rjsf/core';` are now available via `import utils from '@rjsf/utils';`.
+Because of the decoupling of validation from `@rjsf/core` there is a breaking change for all the [validator-based utility functions](https://react-jsonschema-form.readthedocs.io/en/stable/api-reference/utiltity-functions#validator-based-utility-functions), since they now require an additional `ValidatorType` parameter.
+
+If you have built custom fields or widgets that utilized any of these breaking-change functions, don't worry, there is a quick and easy solution for you.
+The `registry` has a breaking-change which removes the unnecessary `definitions` property while adding the new `schemaUtils` property.
```
Comment (heath-freenome) [inline docs/5.x upgrade guide.md:96]: replace `unnecessary` with `deprecated`

File: docs/advanced-customization/custom-templates.md [hunk 3]
```diff
@@ -112,6 +114,7 @@ function CustomFieldTemplate(props) {
 
 render((
   <Form schema={schema}
+        validator={validator}
         FieldTemplate={CustomFieldTemplate} />
```
Comment (heath-freenome) [inline docs/advanced-customization/custom-templates.md:118]: Turn into a single line

File: docs/advanced-customization/custom-templates.md [hunk 4]
```diff
@@ -182,7 +187,7 @@ function ObjectFieldTemplate(props) {
 }
 
 render((
-  <Form schema={schema}
+  <Form schema={schema} validator={validator}
         ObjectFieldTemplate={ObjectFieldTemplate} />
```
Comment (heath-freenome) [inline docs/advanced-customization/custom-templates.md:191]: Turn into a single line

File: docs/advanced-customization/internals.md [hunk 5]
```diff
@@ -71,6 +73,7 @@ const schema = {
 
 render((
   <Form schema={schema}
+        validator={validator}
         onSubmit={onSubmit} ref={(form) => {yourForm = form;}}/>
```
Comment (heath-freenome) [inline docs/advanced-customization/internals.md:77]: Turn into a single line

File: docs/index.md [hunk 6]
```diff
@@ -16,15 +16,22 @@ react-jsonschema-form also comes with tools such as `uiSchema` and other form pr
...
 
-First install the dependency from npm:
+First install the dependencies from npm:
 
 ```bash
-$ npm install @rjsf/core --save
+$ npm install @rjsf/core @rjsf/utils --save
 ```
 
+As of version 5, you will also need to select and install a validator implementation:
```
Comment (heath-freenome) [inline docs/index.md:25]: Such as `@rjsf/validator-ajv6`

File: docs/usage/validation.md [hunk 7]
```diff
@@ -99,30 +116,35 @@ const schema = {
 
 render((
   <Form schema={schema}
+        validator={validator}
         transformErrors={transformErrors} />
```
Comment (heath-freenome) [inline docs/usage/validation.md:120]: Turn into a single line

**Review by nickgros — APPROVED** [review]

**Pull request merged**